packages/integration-sdk-cli/src/bocchi/templates/top-level/src/client.ts.hbs

```diff
@@ -1,53 +1,74 @@
+      }
+      {{/if}}
+    });
+  }
+
+  {{#if (isNotEndpointAuth template.authentication.strategy)}}
+  protected getAuthorizationHeaders(): Record<string, string> {
+    return { {{#each (sanitizeAuthObject template.authentication.authHeaders)}}'{{@key}}': `{{this}}`,{{/each}} };
+  }  
+  {{else}}
+  protected async getAuthorizationHeaders(): Promise<Record<string, string>> {
+    const response = await this.retryableRequest(
+      `${this.baseUrl}{{template.authentication.params.path}}`,
+      {
+        headers: { {{#each (sanitizeAuthObject template.authentication.params.headers)}}'{{@key}}': `{{this}}`,{{/each}} },
+        method: '{{sanitizeHttpMethod template.authentication.params.method}}',
+      {{#if request.params}}
+        body: { {{#each (sanitizeAuthObject template.authentication.params.body)}}{{@key}}: {{this}},{{/each}} }
+      {{/if}}
+      }
+    );
+    return { {{#each (sanitizeAuthObject template.authentication.authHeaders)}}'{{@key}}': `{{this}}`,{{/each}} }
   }
+  {{/if}}
 
   public async verifyAuthentication(): Promise<void> {
     {{#if (isNotEndpointAuth template.authentication.strategy)}}
     // TODO: implement
     await new Promise(() => { throw new Error('Unimplemented: Pick an API call to use for validation!') });
     {{else}}
-    const response = await this.request({
-      url: `${this.baseUrl}{{template.authentication.params.path}}`,
-      headers: { {{#each (sanitizeAuthObject template.authentication.params.headers)}}'{{@key}}': `{{this}}`,{{/each}} },
-      method: '{{sanitizeHttpMethod template.authentication.params.method}}',
-      {{#if request.params}}
-      body: { {{#each (sanitizeAuthObject template.authentication.params.body)}}{{@key}}: {{this}},{{/each}} }
-      {{/if}}
-    });
-    this.authHeaders = { {{#each (sanitizeAuthObject template.authentication.authHeaders)}}'{{@key}}': `{{this}}`,{{/each}} }
+    await this.getAuthorizationHeaders();
     {{/if}}
   }
 {{#each template.steps}}
 
   {{#if (isSingletonRequest response.responseType)}}
   public async get{{pascalCase entity.name}}({{#if parentAssociation}}parentEntity: Entity{{/if}}): Promise<any> {
     const url = `${this.baseUrl}{{{sanitizeUrlPath request.urlTemplate}}}`;
-    return (await this.request({
+    const response = await this.retryableRequest(
       url,
-      headers: { ...this.authHeaders },
-      method: '{{sanitizeHttpMethod request.method}}',
-      body: { {{#each (sanitizeHttpBody request.params)}}'{{@key}}': `{{this}}`,{{/each}} }
-    })).{{response.dataPath}} as {{pascalCase entity.name}};
+      {
+        headers: { ...this.authHeaders },
+        method: '{{sanitizeHttpMethod request.method}}',
+      {{#if request.params}}
+        body: { {{#each (sanitizeHttpBody request.params)}}'{{@key}}': `{{this}}`,{{/each}} }
+      {{/if}}
+      }
+    );
+    return (await response.json() as any).{{response.dataPath}} as {{pascalCase entity.name}};
```

Comment:
```suggestion
    import get from 'lodash/get';
    ...
    return get(await response.json() as any, response.dataPath);
```

Reply:
Your docs seem to indicate that you can pass a `.` delimited path:

```json
{
    "response": {
        "data": [
            { "..." },
            { "..." },
            { "..." },
            "..."
        ]
    }
}
```

<i>Template:</i>

```json
{
  "dataPath": "response.data"
}
```